app/ante/gas_limit.go

```diff
@@ -0,0 +1,41 @@
+package ante
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// defaultMaxGasWantedPerTx is the maximum gas allowed for a single transaction
+const defaultMaxGasWantedPerTx = 50_000_000
```

Comment:
Do we want this to be configurable via the config file?
Asking because that's how Osmosis handles this. It is mostly a local mempool config in their case